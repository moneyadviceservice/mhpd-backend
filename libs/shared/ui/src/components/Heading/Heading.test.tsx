import React from 'react';
import { render, screen } from '@testing-library/react';
import { Heading, H1, H2, H3, H4, H5, H6 } from '.';

describe('Heading component', () => {
  it('renders correctly', () => {
    render(<Heading data-testid="heading">Heading</Heading>);
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  });
  it('renders correctly with custom props', () => {
    render(
      <Heading
        data-testid="heading"
        level="h3"
        color="text-red-700"
        fontWeight="font-[300]"
      >
        Heading
      </Heading>,
    );
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  });
  it('renders with different component', () => {
    render(
      <Heading data-testid="heading" level="h1" component="h2">
        Heading
      </Heading>,
    );
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  });
  it('H1 renders correctly', () => {
    render(<H1 data-testid="heading">Heading</H1>);
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  });
  it('H2 renders correctly', () => {
    render(<H2 data-testid="heading">Heading</H2>);
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  });
  it('H3 renders correctly', () => {
    render(<H3 data-testid="heading">Heading</H3>);
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  });
  it('H4 renders correctly', () => {
    render(<H4 data-testid="heading">Heading</H4>);
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  });
  it('H5 renders correctly', () => {
    render(<H5 data-testid="heading">Heading</H5>);
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  });
  it('H6 renders correctly', () => {
    render(<H6 data-testid="heading">Heading</H6>);
    const heading = screen.getByTestId('heading');
    expect(heading).toMatchSnapshot();
  });
});
