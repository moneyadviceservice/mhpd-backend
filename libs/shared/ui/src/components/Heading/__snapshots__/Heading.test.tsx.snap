// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component H1 renders correctly 1`] = `
<h1
  class="text-4xl md:text-5xl text-gray-800 font-bold"
  data-testid="heading"
>
  Heading
</h1>
`;

exports[`Heading component H2 renders correctly 1`] = `
<h2
  class="text-3xl md:text-4xl text-gray-800 font-bold"
  data-testid="heading"
>
  Heading
</h2>
`;

exports[`Heading component H3 renders correctly 1`] = `
<h3
  class="text-2xl md:text-3xl text-gray-800 font-bold"
  data-testid="heading"
>
  Heading
</h3>
`;

exports[`Heading component H4 renders correctly 1`] = `
<h4
  class="text-xl md:text-2xl text-gray-800 font-bold"
  data-testid="heading"
>
  Heading
</h4>
`;

exports[`Heading component H5 renders correctly 1`] = `
<h5
  class="text-lg md:text-xl text-gray-800 font-bold"
  data-testid="heading"
>
  Heading
</h5>
`;

exports[`Heading component H6 renders correctly 1`] = `
<h6
  class="text-lg text-gray-800 font-bold"
  data-testid="heading"
>
  Heading
</h6>
`;

exports[`Heading component renders correctly 1`] = `
<h1
  class="text-4xl md:text-5xl text-gray-800 font-bold"
  data-testid="heading"
>
  Heading
</h1>
`;

exports[`Heading component renders correctly with custom props 1`] = `
<h3
  class="text-2xl md:text-3xl text-red-700 font-[300]"
  data-testid="heading"
>
  Heading
</h3>
`;

exports[`Heading component renders with different component 1`] = `
<h2
  class="text-4xl md:text-5xl text-gray-800 font-bold"
  data-testid="heading"
>
  Heading
</h2>
`;
